### PR TITLE
Replaces lavaland

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
[Read the hackmd here](https://hackmd.io/@tgstation/BJALmXDth).

This has been an evolving project that I think merits investigation. I am aware that "VR" has been done before. You are free to look through it as I build on it. It is not done at all so I make no guarantees of code quality.

"Mining" becomes an instantiated VR experience. Action is condensed and more recoverable. You can plug in and out, swap pilots, and the damage is linked.

It gives more opportunity for interactions: Players can forcefully disconnect others, save them from certain death, and the experience is highly expandable via map templates.
## Why It's Good For The Game
Much of what the hackmd outlines, and some.
- Miners too disconnected from station
- Mining "action" and mining "job" need to be married
- Bc I like it
## Changelog
:cl:
add: Adds the virtual domain
add: Adds the voidrunner job
del: Removes the shaft miner job role, but not completely
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
Based.

File: code/modules/voidrunning/structures/hololadder.dm [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	if(!neo)
+		balloon_alert(user, "Your mortal connection has been severed!")
+		return
+
+	var/mob/living/carbon/human/occupant = connection.voidrunner_ref?.resolve()
+	if(neo != occupant)
+		balloon_alert(user, "Wrong connection!")
+		return
+
+	balloon_alert(user, "Disconnecting...")
```
Comment (Mothblocks) [inline code/modules/voidrunning/structures/hololadder.dm:47]: none of these balloon alerts should be capitalized. also some are a little long

**Comment by DATA-xPUNGED** [comment]
I just want to understand how this works thematically, how is the station getting ore from a simulation?

**Comment by Mothblocks** [comment]
I do not agree being detached from the station is a bad thing for miners--I think it's a bad thing for NEW content but I think it's fine for miners to fill this niche on their own

**Comment by Pickle-Coding** [comment]
> I just want to understand how this works thematically, how is the station getting ore from a simulation?

Bluespace bro.

**Comment by Mothblocks** [comment]
Also it's a nitpick but "voidrunner" is not in the slightest bit obvious what it does especially compared to "shaft miner"

**Comment by jlsnow301** [comment]
> Also it's a nitpick but "voidrunner" is not in the slightest bit obvious what it does especially compared to "shaft miner"

it's a cyberpunk reference

**Comment by Mothblocks** [comment]
Like if the fix to miners being "detached from the station" is making them sitting duck so sometimes you just die while doing your job and you don't realize it's happening, then have we actually made the game more fun

**Comment by Mothblocks** [comment]
> > Also it's a nitpick but "voidrunner" is not in the slightest bit obvious what it does especially compared to "shaft miner"
> 
> it's a cyberpunk reference

I didn't play Cyberpunk and I'm putting money not enough players have for us to just assume eveyrone knows what it means

**Comment by DustanHache** [comment]
best way to check reception to your little draft is a bit of playtesting. maybe ask to have it testmerged once its more fleshed out.

**Comment by D4C-420** [comment]
Going from "fighting dangerous monsters on a hell world" to "VR dungeon crawling" is a loss in flavor that is not worth it.

**Comment by CCC23** [comment]
Bastards

**Comment by DATA-xPUNGED** [comment]
WHat'll happen to lavaland ghost roles? they're by far the best

**Comment by jlsnow301** [comment]
> best way to check reception to your little draft is a bit of playtesting. maybe ask to have it testmerged once its more fleshed out.

that's my current plan, just wanted to show some progress

**Comment by ChungusGamer666** [comment]
This is so lame, so uninspiring, that I find it hard to put in the effort of writing a detailed essay on why this sucks.

Instead, have this *completely unrelated* system shock gameplay depicting some gameplay of the *widely beloved* cyberspace mechanics.
https://www.youtube.com/watch?v=-Cq7Q7hujjI

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I want to be convinced this is actually beneficial to both parties here, miners and antagonists.

Your design document accepts miners being separated from the station as tacitly true, which I don't agree with. (Again, I think this should be treated as true for *new content* but I think it is acceptable for miners to have this niche)

It also doesn't mention that this turns killing a miner into an engagement that seems fun for neither party (miner just dies, because they don't know people are near them, and traitor kills a sitting duck). There could be forceful ways to solve this like a notification if people are walking around you but it just doesn't sound enjoyable to me at all, and I think my prior concern is still valid here

You mention, for instance, the following problem both on the design document and in Discord:
> Targets off z-level for antagonists are virtually unreachable

I think this is a completely reasonable problem, but I don't think the status quo is improved here at all and that we have better ways to solve this. A *very* forceful one for instance is removing miners from the target pool, which is crummy but also solves that problem better than this. We could also give better antagonistic tools for tracking people down, which would make more areas of the game interesting. Also don't traitors know their target before they pick the objective anyway?

**Comment by Mothblocks** [comment]
Seems fine to test merge tho

**Comment by DustanHache** [comment]
> Like if the fix to miners being "detached from the station" is making them sitting ducks so sometimes you just die while doing your job and you don't realize it's happening, then have we actually made the game more fun for anyone
> 
> image



> I do not agree being detached from the station is a bad thing for miners--I think it's a bad thing for NEW content but I think it's fine for miners to fill this niche on their own

counterpoint to that mothblocks: free golems. they’ve been a point of contention for multiple reasons, including being used as a psudo respawn by people wanting to return to the station after being killed. this is an issue that has arrisen so many times with lavaland roles that the *single* planetbound map in rotation has exactly *one* ghost role for usage unless an admin chooses to spice things up. purging lavaland would free up another Z level for space ruins and ghost roles, and most of the ruins currently exclusive to lavaland and icebox could be ported fairly easily.
another thing to note is that this doesn’t mean the removal or even the exclusion of mining mechanics. just a change to how it is done to keep cargonians as viable interaction for crew and antagonists alike. miners are still going to have access to their gamer gear if someone rips them out of their new VR spessbitcoin miner as *that is still being kept in the game and progression for them according to the hackmd.* they will not be soft targets, but they won’t be inaccessible to everyone either.

**Comment by dieamond13** [comment]
I don't think this is inherently bad but I wouldn't want lavaland removed as it still has a lot of content that should stay.

**Comment by Namwty** [comment]
So couple issues:

Using "bluespace" as solution for stuff, isn't exactly ideal, but that's just my opinion, which isn't worth much.

Not sure if this will really improve interaction, I mean instead of being on different Z-level, miners will be just sitting in one room, but who knows, I could be wrong.

Ghost roles ( Asswalkers and what not, while I do not care for them nor ever played them, there are many who do enjoy them )

I feel like traitors trying to kill a miner would have a bad time ( Can't sabotage outpost comms, can't set up ambush ) + you have to literally kill a miner who is in cargo, which is full of people who are geared to hell. I guess you could add a funny emag function to room... where you could make difficutly go to hell for the miner/voidrunner, would be rather "hitman-esque"

Bounties would maybe need to be reworked for miners.

Relating to bounties, I have suspicion that stuff like lavaland vegetation would be scarce. Another big issue is that I imagine geysers would be gone aswell ( chemist tend to use wittel and so on ).

Repair of said room would be a thing aswell, lack of power, C4 and other explosives, bad atmos from a flood and what not, or even simple meteors could make it so voidrunners/miners can't do their job, leaving station piss poor.

Most fauna/megafauna would have issues to work with this and would require severe changes, but you already mentioned it.

I'm a miner main tbf, and even had played on asteroid, while changing this would be risky and generally just sounds like would add shit tons of problems and exploits. I don't mind something new, if actually it is being checked over and what not.

But hey, this would remove chasms... right?

Jungle generation and what not would be nice tho, could actually add more flavor.

**Comment by Mothblocks** [comment]
> counterpoint to that mothblocks: free golems. they’ve been a point of contention for multiple reasons, including being used as a psudo respawn by people wanting to return to the station after being killed.
this is an issue that has arrisen so many times with lavaland roles that the single planetbound map in rotation has exactly one ghost role for usage unless an admin chooses to spice things up. purging lavaland would free up another Z level for space ruins and ghost roles, and most of the ruins currently exclusive to lavaland and icebox could be ported fairly easily.

I'm not sure why this is a counterpoint to me. I'm talking about specifically the shaft miner role being off station is fine to me, but I do not like the other stuff that is or tries to push people in that direction

**Comment by Krypandenej** [comment]
Just reflavor this into a sci job, maybe to make up for the loss of expeditionary corps.

**Comment by B00tz** [comment]
Really ambitious PR here but I got a few questions that don't seem to be answered by the hackmd.

1. What do you think should happen to all of the other stuff associated with lavaland that isn't tied to the shaft miners? Auxiliary Base Construction
Gulag
Geysers 
Are all of these just gonna be ported over to the VR system?
2. You mentioned how antags could still get megafauna onto the station but what do you think that would look like? Let's say I emag the sleeper, what do you think should happen?
3. This one isn't really a question but you said you joined in 2021 but even before that there was something similar with the VR sleepers. Now I don't know all the ins and outs of coding or just how stark the difference is between the power we got now compared to back then but it might be a good idea to take a look at some of the reasons given in the removal pr #43832

**Comment by DustanHache** [comment]
> I'm not sure why this is a counterpoint to me. I'm talking about specifically the shaft miner role being off station is fine to me, but I do not like the other stuff that is or tries to push people in that direction

it’s a counterpoint because it comes with the territory currently associated with miners. part of the issue derives from the fact that *miners interact with them.* it is so frequent of a headache that both coder and admin alike have had to implement policies and changes to prevent lavaland roles from being taken back to station. I cannot begin to count the number of times i have seen ashwalker eggs brought back to the station *by miners* nor do I ever wish to see it happen again. 

> Really ambitious PR here but I got a few questions that don't seem to be answered by the hackmd.
> 
> 1. What do you think should happen to all of the other stuff associated with lavaland that isn't tied to the shaft miners? Auxiliary Base Construction
>    Gulag
>    Geysers
>    Are all of these just gonna be ported over to the VR system?

I would assume gulag would be converted to a forced VR sim of sorts. Geysers might be moved to space, auxil base construction would likely be converted to the room/staging ground for the whole VR endeavor. perhaps a window in to the miner’s VR and in turn a key as to how megafauna escape?

**Comment by xprtdrt** [comment]
this seems like a cool concept, but not as a total lavaland replacement
If you still enjoy the vr idea, you could use it as a sort of training room for shaft miners, where they get some preset gear and are facing a lavaland boss in it. it would make practicing to kill these bosses on lavaland itself a lot easier, since there wouldn't be a permadeath risk. maybe when they kill the boss, they could get a bunch of miner points to incentivize it? that sounds like a fun idea to me, and even experienced miners might use it when they want a bunch of mining points early on

**Comment by AlamoTurtle** [comment]
So you want to remove the entirety of lavaland and its efforts pumped in because it feels "isolated" and replace it with some boring VR with little to no "repercussions"? It's detached from the station, yes, but anyone can go to lavaland. Anyone, via shuttle. Mining, while detached from the station, still returns to cargo to bring their haul back, granted they do not die, and should they choose to, they may stop mining for a bit to chill on station and show off their loot. This entire PR feels like it's dumbing down gameplay to take away the vast playground miners have to shackle them on-station. As I played TG, I have learned that Shaft Miner is a really friggin fun role.

If you play on Goonstation, their mining is... "hook an asteroid, whack it a bit, start dying from radioactive rocks in the most boring sense ever, and maybe if you're competent you can have a funny industrial mining suit before your shift ends or you die", which is kind of what your premise seems to go off of: Generate a random room and get shit to mine and maybe some enemies to fight. I... wholeheartedly disagree in this PR. This is just a sake of removing things for "something new" instead of adding it a completely new feature such as some new volunteer feature with "drone exploration" or a "VR assistance program" to send in a holo-you in Lavaland to potentially assist miners from the safety of station as an opt-in to just going down yourself, with severe limitations. It'd be something paramedics could enjoy while also giving lavaland miners the best of both worlds depending on which approach they like.

Basically, removing an entire feature of the game because you think holodeck VR is a funny idea is the worst way to go about this. Sorry to say it, but I don't think I'd ever touch shaft miner or this content whatsoever and many who share the same sentiment would mean materials have to be sourced from those who want to undergo the slog, or with space mining, which would personally be more fun and more rewarding to me than the VR chamber.

**Comment by Mothblocks** [comment]
> it’s a counterpoint because it comes with the territory currently associated with miners. part of the issue derives from the fact that miners interact with them. it is so frequent of a headache that both coder and admin alike have had to implement policies and changes to prevent lavaland roles from being taken back to station. I cannot begin to count the number of times i have seen ashwalker eggs brought back to the station by miners nor do I ever wish to see it happen again.

It's not a counterpoint at all bceause I think free golems shouldn't exist is my point lol

**Comment by DustanHache** [comment]
> > it’s a counterpoint because it comes with the territory currently associated with miners. part of the issue derives from the fact that miners interact with them. it is so frequent of a headache that both coder and admin alike have had to implement policies and changes to prevent lavaland roles from being taken back to station. I cannot begin to count the number of times i have seen ashwalker eggs brought back to the station by miners nor do I ever wish to see it happen again.
> 
> It's not a counterpoint at all bceause I think free golems shouldn't exist is my point lol

i suppose you think the derilict and charliestation shouldnt exist either then?

**Comment by YehnBeep** [comment]
What if something like this was instead an evolution of drone exploration?

**Comment by ChungusGamer666** [comment]
> What if something like this was instead an evolution of drone exploration?

He doesn't know rule 1 of TG contributions, every meaningful addition must also remove something else from the game.

**Comment by AlamoTurtle** [comment]
> What if something like this was instead an evolution of drone exploration?

Exactly, there's plenty of mechanics able to be touched up and if this feature really wanted to be implemented without the destruction of lavaland in its entirety, a simple optional room with about 3-5 holopads to stand on and interact with to "challenge the Holo-Gauntlet" with some specific gear you can configure beforehand would be far preferable imo, bringing back the desired loot if you're successful and killing you via some holo damage or whatever if not. There are already mechanics that the holodeck has for damage sustained via holodeck.

**Comment by SpotRock** [comment]
I hope this gets added!

**Comment by optimumtact** [comment]
good change i've never met a mining player who was likeable as a person.

**Comment by ChungusGamer666** [comment]
> good change i've never met a mining player who was likeable as a person.

so you're a miner main?

**Comment by Ticklezcat** [comment]
The easiest way I can think of to make it likable for players would be to have it as an alternate to lavaland rather than a replacement, similar to the Ice moon. You could also make it even more attractive by maybe putting it on a currently discontinued but popular station like how Icebox is just Box with lavaland replaced, maybe Kilo or Pubby. A station revamp in general would also give room for some theming to make the VR aspect fit better.

**Comment by CoiledLamb** [comment]
> What if something like this was instead an evolution of drone exploration?

god i wish

**Comment by CCC23** [comment]
> > What if something like this was instead an evolution of drone exploration?
> 
> god i wish

If you want more explo drone just do it its one of the most simplistic jobs

**Comment by wesoda25** [comment]
>Targets off z-level for antagonists are virtually unreachable

Disagree, personally I think miners pose a unique challenge for an antagonist that can be pretty fun, on occasion, to deal with. Public Mining Access + the miner culture of taking a GPS results in them not being _that_ difficult to find. The additional dangers of lavaland + how strong miners can be requires you to really think outside the box on how to get your hands on them, whether you're trying to kill them or just want their family heirloom. 

In reference to this specific issue and assuming this isn't merged, I'd say it'd be a good idea to look into increasing miner reliance upon the station, if not the outpost. Limiting the number of bluespace capsules available to one or two would be a good way of doing this, it'd significantly decrease the area a miner could possibly be in which would make them a bit less of a pain as a target.

**Comment by CliffracerX** [comment]
There are problems with this and they are many.

- "Because I like it" is not a real justification for how this improves the game.
- Making new Z-levels is ill performant at the best of times, and giving every would-be miner their own is an even bigger drain.
- This detaches the existing shaft miners from station gameplay even more by entirely removing them from reality.
- SS13 is a metal deathtrap masquerading as a futuristic station.  Lavaland fits that theme.  This doesn't.

As a whole, the justification for removing Lavaland is to bring miners back to the station, and to make them part of the typical SS13 gameplay loop - but having them jack in and become oblivious to the station does the exact opposite.  There's no reasonable thematic explanation for how a virtual world can grant you real rewards - the ore has to come from somewhere - and if you make it entirely into a recreational area, you're back to square one: mining has to happen somewhere, and by its nature it's going to be off-station and away from the primary antag loop.  Short of removing mining entirely and just letting Cargo order all types of ores, *someone* has to be off the station to get those materials, and that's okay.

**Comment by SethLafuente** [comment]
This is just awfull I dont even have to explain why is stupid anyways 

What about icebox is the map gona be removed since its essentially station in a lavaland type place 

You can introduce VR without removing lavaland

**Comment by itseasytosee** [comment]
I love it 👍

**Comment by Sheits** [comment]
is this how bitcoin mining works irl

**Comment by Capsandi** [comment]
I don't think reintroducing VR would be prudent considering how many edge cases the last iteration had, but if this gains traction and you need help mapping an initial set of modules I am in the discord.

**Comment by MTandi** [comment]
>SS13’s core appeal lies in the interaction and ensuing chaos between crew members.

Totally disagree that everything should be focused on multiplayer aspect. 

Some people love to do something in isolation interacting with just a few or no players. Letting them do this, while also providing benefit for the station in form of minerals is a genius idea and lavaland is one of the best things about ss13.

**Comment by rasepretep** [comment]
Putting aside the faulty argument that miners are unreachable for antags - faulty because ambushing and killing a miner on lavaland / in mining base is one of the easiest kills which all but guarantees that you get away with it, **provided the antag preps for it**, Lavaland removal would screw with other things more.

**Removing Lavaland also removes:**
Lavaland ghostroles, including the Syndicate Outpost, which is a great asset for traitors to work with, if the syndies spawn
Escaping to Lavaland in the event of a stationwide disaster, whether by mining shuttle or launching pods during Delta
A place for people who want to go down to chill and build a cool house
Public mining
Gulag
Antags making a secret base on Lavaland, or escaping there to avoid arrest
AUX Base platform
Any enjoyment a miner might have from having to survive in a barren, hostile environment where a misstep might mean certain death. This is why I started playing Shaft Miner, and why I stuck with it as a miner main all this time. The thrill of having to survive, not knowing what I'll see when I come back to station, not knowing if someone has set up a trap for me in mining base, or if that assistant running around with a pickaxe is actually a traitor waiting for the right moment to kill me.

Miners going down and never coming back up to station is a player issue, not a design issue, same as how some curators simply leave the station and stay in space the whole round, hoovering up ruins loot. A typical miner will be up on station very often, especially if they aren't really into hunting megafauna, or simply get bored of the grind, and there are a lot more miners like that, than there are unga-miners that only exist to fight monsters.

On top of that, this makes killing miners a lot more difficult, because now they are on station and in a room from which you can easily hear gunshots, so people will more often opt for bombing the room, which greatly cripples ore production, if not halting it entirely.

I'm really not sure what to think about the argument that it promotes crew interaction. Sitting in a chair jacked into VR doesn't really feel like it generates more interaction beyond being vulnerable to anyone walking up and doing anything they please.

Furthermore, how is this supposed to work with Icebox mining, would we be simply removing the areas around the station there?

Shaft Mining being hostile to newbies, carrying a steep learning curve, is a problem, but this isn't the way to solve that.

Overall, this entire PR feels like the primary reasons for it is general dislike for Lavaland even existing, the misconception that Miners are immortal and their gear OP (that is a miner robustness and/or player powergaming issue), and an excuse to put in a vague reference to a video game because the word "voidrunner" sounds cool.

Mining is headed towards a necessary rework from the ground up, we all know that. But this isn't the right way to do it.

**Comment by Striders13** [comment]
replace cargo drones with this

**Comment by Likteer** [comment]
> We could also give better antagonistic tools for tracking people down

Yes, please!

File: code/modules/voidrunning/structures/netchair.dm [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	for(var/obj/machinery/quantum_server/server as anything in oview(7))
+		if(istype(server, /obj/machinery/quantum_server))
+			src.server = server
+			return TRUE
+
+	return FALSE
+
+/// Resolves a path to an outfit.
+/obj/structure/netchair/proc/resolve_outfit(text)
+	var/path = text2path(text)
```
Comment (DrPepperIsSpicy) [inline code/modules/voidrunning/structures/netchair.dm:169]: This doesn't seem safe. I would use a list instead or check specifically if the outfit is in /outfit/job

**Comment by SomeoneYouProbablyKnow** [comment]
Absolutely based swing taken. Keep at it, king

**Comment by TheVekter** [comment]
I'm definitely for the general idea of finding a replacement for Lavaland. Most rounds either result in every miner dying (usually to a megafauna) or them surviving but still only mining for maybe five minutes, so the station ends up not getting enough resources to actually do anything beyond the basic required research.

I don't like that shaft miners don't actually _mine_ for any significant amount of time. I had suggested ages ago that we split the jobs off into two, one strictly for mining and the other for exploration and megafauna killing (or just give that aspect to curator) but it wasn't very popular.

That being said, I'm not yet convinced that this does what it thinks it does, but you've got time to cook, so we'll see how this ends up.

**Comment by atlasle** [comment]
I just want to say, if this gets closed, please don't throw all of this away

**Comment by Rex9001** [comment]
If I understand this correctly every room is 15x15 which seems a tad bit small, I like this concept since it could allow for more biomes and what not but I am not convinced that kicking a station with little power whilst its down by blocking mineral generation is a good idea.

**Comment by ilikesaying** [comment]
Where will escape pods go when they’re launched pre-shuttle-launch?

**Comment by Toastgoats** [comment]
Wouldn't it make more sense to just add mining drones that miners pilot stationside instead of axing a massive part of the game in an overwhelmingly unpopular way?

**Comment by ChungusGamer666** [comment]
> I feel like this is change for the sake of change.

image

**Comment by OverwatchVoice** [comment]
There are so many mental gymnastics to unpack here I'm getting a headache trying to formulate an opinion

> What if something like this was instead an evolution of drone exploration?
I second this idea though.

**Comment by Fikou** [comment]
> Burden on mappers to include this room

this is an understatement.
fugly 15x15 rooms are terrible to include (see plumbing, and that one at least has decal decoration and machinery around), and id rather we not have on of those on every map

**Comment by Fikou** [comment]
also how the fuck are you getting rocks from vr

**Comment by SomeoneYouProbablyKnow** [comment]
> also how the fuck are you getting rocks from vr

if you mine in the game you mine in real life

**Comment by dwasint** [comment]
i feel like this should just replace gateway missions in some way instead of gutting lavaland, the main issue is its gonna be a pain to rp kill voidrunners, also voidrunners have no incentive to do their job if they are antags what so ever as it makes them incredibly vulnerable. Ghost roles would need to be completely changed or removed to facilitate this. Its a cool concept that probably should be standalone as something like an exploration situation

**Comment by SweptWasTaken** [comment]
> also how the fuck are you getting rocks from vr

Maybe this idea could be slightly reworked to make the "VR" part of miners them controlling remote mining mechs/vehicles on lavaland, but plugged into VR, that could be upgraded over time through science or exploration. The materials could be teleported back using a collection module, or maybe someone could pilot around a collection skiff. 

Also kind of a cursed idea but you could implement a Pacific Rim style Kaiju-killing mech where one person controls the movement and another controls the combat/mining.

**Comment by AtlasRules** [comment]
This is cool and all but like,

Why mining? Why not science? This fits way better for science, netrunning and all that.

I hate to ideasguy but honestly, if you want to crib from a game for a mining replacement, rip off Deep Rock Galactic. Better yet, port the NSV asteroid magnet  or something.

**Comment by AtlasRules** [comment]
> > also how the fuck are you getting rocks from vr
> 
> Maybe this idea could be slightly reworked to make the "VR" part of miners them controlling remote mining mechs/vehicles on lavaland, but plugged into VR, that could be upgraded over time through science or exploration. The materials could be teleported back using a collection module, or maybe someone could pilot around a collection skiff.
> 
> Also kind of a cursed idea but you could implement a Pacific Rim style Kaiju-killing mech where one person controls the movement and another controls the combat/mining.
> 
> Additional ideas-guying, antags could hijack mechs or collection vehicles and teleport timed explosives or w/e back to the station.

I think there's a mech like that in some places, where one guy controls additional weapons and the other controls the mech's movement.

**Comment by Dawnseer** [comment]
> > > also how the fuck are you getting rocks from vr
> > 
> > 
> > Maybe this idea could be slightly reworked to make the "VR" part of miners them controlling remote mining mechs/vehicles on lavaland, but plugged into VR, that could be upgraded over time through science or exploration. The materials could be teleported back using a collection module, or maybe someone could pilot around a collection skiff.
> > Also kind of a cursed idea but you could implement a Pacific Rim style Kaiju-killing mech where one person controls the movement and another controls the combat/mining.
> > Additional ideas-guying, antags could hijack mechs or collection vehicles and teleport timed explosives or w/e back to the station.
> 
> I think there's a mech like that in some places, where one guy controls additional weapons and the other controls the mech's movement.

Yeah, we have that mech. It's the Savannah

**Comment by SomeoneYouProbablyKnow** [comment]
> Why mining? Why not science? This fits way better for science, netrunning and all that.

Science is next 🔫

**Comment by AtlasRules** [comment]
> > Why mining? Why not science? This fits way better for science, netrunning and all that.
> 
> Science is next 🔫

so use this for science

**Comment by optimumtact** [comment]
lets not waste anymore time than has already been wasted.
for a safe bet any time you're about to wholesale "replace" someething, make sure you get at least 2 headcoder's signoff first.

**Comment by atlasle** [comment]
> I'm definitely for the general idea of finding a replacement for Lavaland. Most rounds either result in every miner dying (usually to a megafauna) or them surviving but still only mining for maybe five minutes, so the station ends up not getting enough resources to actually do anything beyond the basic required research.
> 

This can be fixed without removing lavaland

**Comment by jlsnow301** [comment]
> lets not waste anymore time than has already been wasted.
> 
> for a safe bet any time you're about to wholesale "replace" someething, make sure you get at least 2 headcoder's signoff first.

Thanks, will be reopening this

**Comment by jlsnow301** [comment]
> My main problem with this line of rework is if they are in VR for the entire round then the interaction problem isn't ...

Nice to see you here bacon. This has evolved well beyond the hackmd with more design consideration, including this concern.

In short,
- Dungeons are ~5minutes long
- Server gives downtime after completing a map- ~~voidrunners~~ bitminers can go get a drink / heal up / return to action
- I'm not opposed to random generation, but as of yet I'm making the maps by hand and porting existing ruins.
- I've made it super easy to make new instances and it's fully UI'd

It's not totally unlike bee exploration but the key here was leaving some ties back to the station, which I think "vr" solves perfectly

**Comment by Striders13** [comment]
> It's not totally unlike bee exploration but the key here was leaving some ties back to the station, which I think "vr" solves perfectly

I think you can solve the issue of miners beeing too vulnerable by giving them a vr headset and letting them pace around a phyisical room on station, but only they would be able to see the contents of said room. Would make it easy to spot intruders.
I'm not sure how hard it'd be to code that tho, im just the ideasguy.

**Comment by jlsnow301** [comment]
> I'm not sure how hard it'd be to code that tho, im just the ideasguy.

you should try it.

**Comment by Yobrocharlie** [comment]
So this was the birth of bitrunning,
This freaking brain fart of a terrible PR…

interesting

**Comment by Yobrocharlie** [comment]
> Also working on lavaland replacements but since bee uses supercruise it's a lot easier for us so feel free to dm me any ideas (Bee has randomly generated lazy dungeons with z clearing so I can talk a bit about the flaw and challenges of that kind of system). My main problem with this line of rework is if they are in VR for the entire round then the interaction problem isn't solved. Jobs on the station have isolation issues (xenobiology, virology, genetics etc.)

Bacon my beloved,
Same person who says they love using the sleepy pen truly somebody who cares about engaging gameplay <3

also man beestation has really hit the fucking fan

**Pull request closed without merging**